components/camel-google-pubsub/src/main/java/org/apache/camel/component/google/pubsub/GooglePubsubComponent.java

```diff
@@ -129,6 +132,15 @@ private Publisher buildPublisher(String topicName) throws IOException {
             CredentialsProvider credentialsProvider = NoCredentialsProvider.create();
             builder.setChannelProvider(channelProvider).setCredentialsProvider(credentialsProvider);
         }
+        if (googlePubsubEndpoint.isMessageOrderingEnabled()) {
+            builder.setEnableMessageOrdering(true);
+            if (StringHelper.trimToNull(googlePubsubEndpoint.getPubsubEndpoint()) != null) {
+                builder.setEndpoint(googlePubsubEndpoint.getPubsubEndpoint());
```

Comment:
I am bit confused here, why do you need to override the endpoint here if the message ordering is enabled? And also it looks the endpoint could be overriden [here](https://github.com/apache/camel/pull/4877/files#diff-fab91f6f43de3fd75a1134f3fe2fda9b52f3d78a45336ab56a758eae05fba1edR129) and hence I don't think this option is needed.

Reply:
From what I see the endpoint is used by the  description = "Endpoint to use with local Pub/Sub emulator."
when running unittests. Also it set's up a NoCredentialsProvider so I guess this is not a 'real' runtime usable property.

Reply:
I see. My problem here is that, we will end up with two similar endpoint options which could be a bit confusing to the user. Hence, can we somehow refactor this to only have one endpoint option to be overriden (emulator and API endpoint)? Also, I don't think overriding the endpoint should be only possible when the message ordering is enabled.

Reply:
@oscerd Would be good if we could improve this later so that the emulator uses the pubsubEndpoint. 
@omarsmak Agree overriding the endpoint should be possible even without message ordering.

Reply:
Okay, then for now we can just enabling the endpoint override without the message ordering. And then later we can do the endpoint refactor (to have only one endpoint option to configure instead of two). Would be great if you can create a JIRA for the proposed improvement.

Reply:
enabled endpoint override, and created https://issues.apache.org/jira/browse/CAMEL-16036 to improve the component later.